system/lib/wasmfs/file_table.h

```diff
@@ -87,4 +89,17 @@ class FileTable {
   // There is only ever one FileTable in the system.
   static Handle get();
 };
+
+// Root Directory is a singleton that will lazily initialize default files and directories.
+class RootDirectory : public Directory {
```

Comment:
Since the only functionality this adds over `Directory` is the initialization of `/dev`, but that initialization would be possible to do via the public `Directory` API, I wonder if it would make sense to not have a special `RootDirectory` subclass, but rather perform the initialization via the `Directory` API when the root directory is initialized.